File: 66-plus-one/66-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();
        int i = n-1;
        bool carry = false;
        while(i>-1){
            if(digits[i]+1<10){
                digits[i]++;
                return digits;
            }else{
                digits[i] = 0;
                i--;
                carry = true;
            }
        }
        vector<int> v;
        if(carry)v.push_back(1);
        for(int i = 0;i<n;i++)v.push_back(digits[i]);
        return v;
    }
};

File: 66-plus-one/66-plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        n = len(digits)
        i = n-1
        carry = False
        while(i>-1):
            if(digits[i]+1<10):
                digits[i]+=1
                return digits
            else:
                digits[i] = 0
                i-=1
                carry = True
        if(carry):
            digits.insert(0,1)
        return digits
        